src/components/index.ts

```diff
@@ -44,6 +44,9 @@ enum Size {
     l = 'l',
     xl = 'xl',
     xxl = 'xxl',
+    regular = 'regular',
+    big = 'big',
+    huge = 'huge',
```

Comment:
Attention par ce que le composant `<Icon>` utilise `Size` en type de props et donc ça veut dire que maintenant il est censé accepter `Size.regular` en input. 

Soit on part sur l'utilisation d'un seul `enum` comme celui là, soit on crée un nouvel enum pour `tiny|regular|big|huge` qui sera utilisé ici, pour le `ListItemSize` et pour la `Dialog` plus tard.

Si on part sur l'utilisation d'un seul `enum`, il faudrait s'assurer que tous les props des composants utilisent bien un subset de `Size` via un union type. (ex: `type AvatarSize = Size.xs | Size.s | Size.m | Size.l | Size.xl;`)

Reply:
Bonne remarque, je sais pas ce qui est le mieux de faire.
@GuillaumeNachury ?

Reply:
Un union type avec un type guard ça peut être pas mal.

`
type AvatarSize = Size.xs | Size.s | Size.m | Size.l | Size.xl;

function isAvatarSize(x: string): x is AvatarSize {
  return x === 'tiny' || x === 'regular' || x ===  'big' || x === 'huge';
}
export { AvatarSize, isAvatarSize };
`

Reply:
Ça me paraît mieux de partir sur deux enums: `Size` pour les tailles "fixes" (`s`, `m`, ...) et par exemple `Proportion` pour les tailles "variables" (`regular`, `big`, ...)

Reply:
Faut voir si on n'a besoin de gérer les 2 dans un même composants. Si oui, on peut partir sur des intersection types avec des type guards.

Reply:
Pour en avoir pas mal discuté avec Mathias, `xs`, `s` ... et `tiny`, `huge` représentent l'information de taille. Du coup ca fait du sens qu'ils se retrouve tous dans cette enum.
Il me parait effectivement logique que les composants utilise un type adapté à leur besoin et non pas se baser sur une enum.
Je vais partir sur un type spécifique